### PR TITLE
Call `WizardScene::middleware()` on enter

### LINKED ISSUE #1333: Invoking ctx.scene.enter() on a WizardScene doesn't actually start the scene until the next update
It does create the state within the session, and sets the current scene within the state, but doesn't set the cursor, and more importantly, never even enters the first step until the next update.

I tracked this down to the following code within SceneContextScene.enter():

```javascript
const handler =
      'enterMiddleware' in this.current &&
      typeof this.current.enterMiddleware === 'function'
        ? this.current.enterMiddleware()
        : this.current.middleware()
```

This predicate is <i>always</i> true: 
```javascript
'enterMiddleware' in this.current &&
      typeof this.current.enterMiddleware === 'function'
```

And therefore handler is always set to `this.current.enterMiddleware()`, which returns `this.current.enterHandler`, which is, in turn, a passthru composer that simply wastes the current request doing nothing.

I am currently working around this by setting:
```javascript
Scenes.WizardScene.prototype.enterMiddleware = Scenes.WizardScene.prototype.middleware;
```

Just before defining my `WizardScene`s.

### PR DESCRIPTION
**Pull request by Loskir**
# Description

Hopefully fixes #1333 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

typecheck and ava

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by wojpawlik — APPROVED** [review]
Thank you for saving the day, this does fix #1297.

Please move the new tests to a separate file, then I'll merge.

File: src/scenes/wizard/index.ts [hunk 1]
```diff
@@ -55,4 +57,8 @@ export class WizardScene<
       },
     ])
   }
+
+  enterMiddleware() {
+    return compose([this.enterHandler, this.middleware()])
```
Comment (wojpawlik) [inline src/scenes/wizard/index.ts]:
```suggestion
    return Composer.compose([this.enterHandler, this.middleware()])
```

File: src/scenes/wizard/index.ts [hunk 2]
```diff
@@ -5,6 +5,8 @@ import Composer from '../../composer'
 import Context from '../../context'
 import SceneContextScene from '../context'
 
+const { compose } = Composer
+
```
Comment (wojpawlik) [inline src/scenes/wizard/index.ts]:
```suggestion
```

**Comment by wojpawlik** [comment]
Would you mind moving the new tests to `test/scenes.js`?

**Comment by Loskir** [comment]
Good idea

**Pull request merged**